economy/models.py

```diff
@@ -132,7 +137,7 @@ class Purchase(models.Model):
         related_name='verified_purchases',
         on_delete=models.DO_NOTHING
     )
-    signed_off_time = models.DateTimeField(auto_now_add=True)
+    signed_off_time = MonitorField(monitor='signed_off_by')
```

Comment:
MonitorFields will as default store the current datetime, and update it whenever the monitored attribute is changed. We want to store as null until signed_off_by is set for the first time.
```suggestion
    signed_off_time = MonitorField(monitor='signed_off_by', null=True, default=None)
```